hostwatch/status_syncer.go

```diff
@@ -0,0 +1,57 @@
+package hostwatch
+
+import (
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher-metadata/metadata"
+	"github.com/rancher/kubernetes-agent/kubernetesclient"
+)
+
+var (
+	maxRetryCount int = 3
+)
+
+func statusSync(kClient *kubernetesclient.Client, metadataClient metadata.Client) error {
+	hosts, err := metadataClient.GetHosts()
+	if err != nil {
+		return fmt.Errorf("Error reading host list from metadata service: [%v]", err)
+	}
+	for _, host := range hosts {
+		mNodeStatus := host.State
+		switch mNodeStatus {
+		case ActivatingState:
+			err = cordonUncordon(host, kClient, metadataClient, false)
```

Comment:
looks like we might lose the error if several hosts are being processed, and one of them fails, and another passes